pkg/reconciler/broker/trigger/trigger.go

```diff
@@ -153,9 +153,18 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, t *eventingv1.Trigger) p
 
 func (r *Reconciler) resolveDeadLetterSink(ctx context.Context, b *eventingv1.Broker, t *eventingv1.Trigger) error {
 	// resolve the trigger's dlq first, fall back to the broker's
-	for _, delivery := range []*eventingduckv1.DeliverySpec{t.Spec.Delivery, b.Spec.Delivery} {
+	if t.Spec.Delivery != nil && t.Spec.Delivery.DeadLetterSink != nil && t.Spec.Delivery.DeadLetterSink.Ref != nil &&
+		t.Spec.Delivery.DeadLetterSink.Ref.Namespace == "" {
+		t.Spec.Delivery.DeadLetterSink.Ref.Namespace = t.GetNamespace()
+	}
+
+	var parent interface{} = t
+	for i, delivery := range []*eventingduckv1.DeliverySpec{t.Spec.Delivery, b.Spec.Delivery} {
 		if delivery != nil && delivery.DeadLetterSink != nil {
-			dlqURI, err := r.uriResolver.URIFromDestinationV1(ctx, *delivery.DeadLetterSink, b)
+			if i > 0 {
+				parent = b
+			}
+			dlqURI, err := r.uriResolver.URIFromDestinationV1(ctx, *delivery.DeadLetterSink, parent)
```

Comment:
I'd also name this `deadLetterSink`  see here: https://github.com/knative/eventing/blob/2fcbc0ac500dc06433a851e242fa57a23dedbdfa/pkg/reconciler/subscription/subscription.go#L278

similar code, that also propagates down to status